addons/mrp/models/product.py

```diff
@@ -113,6 +113,33 @@ def _compute_mrp_product_qty(self):
                 continue
             product.mrp_product_qty = float_round(mapped_data.get(product.id, 0), precision_rounding=product.uom_id.rounding)
 
+    def _get_bom_kits(self):
+        products = self.filtered(lambda p: p.type != 'service' or p.product_tmpl_id.type != 'service')
+        if not products:
+            return {}
+
+        def get_bom(kits, product_id, produt_tmpl_id):
+            template_bom = kits.get((False, produt_tmpl_id), {'sequence': float('inf')})
+            product_bom = kits.get((product_id, produt_tmpl_id), {'sequence': float('inf')})
+            return template_bom.get('bom') if template_bom['sequence'] < product_bom['sequence'] else product_bom.get('bom')
```

Comment:
I think I forgot to check sorting in my own PR
https://github.com/odoo/odoo/pull/68229#pullrequestreview-618484779